server/lib/config/nginx.conf

```diff
@@ -82,6 +82,10 @@ http {
         error_page   500 502 503 504  /50x.html;
         error_page   597              @maintenance;
 
+        if ($request_uri ~* "/api/.*/upload/") {
+            client_max_body_size     100G;
+        }
```

Comment:
Does this work across the redirect to the error page??

Reply:
I don't know, but I expect it to work across the `/404.html` and `/50x.html` _locations_, because the `server` config is (supposedly) processed before the `location` configs and it is processed only once.

However, Mr. Jenkins says that Nginx doesn't like it:
```
nginx: [emerg] "client_max_body_size" directive is not allowed here in /etc/nginx/nginx.conf:86
```
😡

Reply:
Apparently `client_max_body_size` is not permitted in an `if` block.  Moreover, the argument to it apparently must be a literal -- it won't accept a variable (unless you're running a fork of Nginx, like Cloudflare apparently does) -- so there doesn't seem to be any way to set it conditionally, other than via separate `location` blocks.

So, either I repeat the directive in all the `locations` where I want it, or I move it to a more globalish scope.  😠

Reply:
With the three directives in their respective locations, it works as hoped:  after 2.5min, `pbench-results-push` prints out:
```
HTTP Error status: 504, message: <!DOCTYPE html PUBLIC "...
```
So, that much is good.

Now, @dbutenhof, are you good with the repeated directives, or should I combine them into a globalish one?

Reply:
Sigh. We don't actually want a limit of 100G for the error pages, and I hate the duplication. Can we just move it up to the `http` or at least `server` block? Note that if you really are worried about limiting data sent to `/dashboard`, you could just override the default with another `location`; though that hardly seems worth worrying about.